src/applications/mhv-medical-records/reducers/blueButton.js

```diff
@@ -0,0 +1,271 @@
+    },
+    primaryNextOfKin: {
+      name: item.nextOfKinName || 'None recorded',
+      address: {
+        street: `${item.nextOfKinStreet1} ${item.nextOfKinStreet2 ||
+          ''}`.trim(),
+        city: item.nextOfKinCity,
+        state: item.nextOfKinState,
+        zipcode: item.nextOfKinZipcode,
+      },
+      phone: item.nextOfKinHomePhone || 'None recorded',
+    },
+    emergencyContact: {
+      name: item.emergencyName || 'No information reported',
+      address: {
+        street: `${item.emergencyStreet1} ${item.emergencyStreet2 ||
+          ''}`.trim(),
+        city: item.emergencyCity,
+        state: item.emergencyState,
+        zipcode: item.emergencyZipcode,
+      },
+      phone: item.emergencyHomePhone || 'No information reported',
+    },
+    vaGuardian: 'No information reported',
+    civilGuardian: 'No information reported',
+    activeInsurance: 'No information reported',
+  };
+};
+
+/**
+ * Convert the FHIR vaccine resource from the backend into the appropriate model.
+ * @param {Object} vaccine a FHIR vaccine resource
+ * @returns a vaccine object that this application can use, or null if the param is null/undefined
+ */
+
+export const convertMilitaryService = data => {
+  if (typeof data === 'undefined' || data === null) {
+    return null;
+  }
+  return {
+    id: data.id,
+    name: data.vaccineCode?.text,
+  };
+};
+
+/**
+ * Convert the FHIR vaccine resource from the backend into the appropriate model.
+ * @param {Object} vaccine a FHIR vaccine resource
+ * @returns a vaccine object that this application can use, or null if the param is null/undefined
+ */
+
+export const convertAccountSummary = data => {
+  if (typeof data === 'undefined' || data === null) {
+    return null;
+  }
+  // Extract necessary fields
+  const { facilities, ipas } = data;
+
+  // Map facilities
+  const mappedFacilities = facilities.map(facility => ({
+    facilityName: facility.facilityInfo.name,
+    stationNumber: facility.facilityInfo.stationNumber,
+    type: facility.facilityInfo.treatment ? 'Treatment' : 'VAMC',
+  }));
+
+  // Extract user profile details
+  const authenticationInfo =
+    ipas && ipas[0]
+      ? {
+          source: 'VA',
+          authenticationStatus: ipas[0].status || 'Unknown',
+          authenticationDate: new Date(
+            ipas[0].authenticationDate,
+          ).toLocaleDateString('en-US', {
+            year: 'numeric',
+            month: 'long',
+            day: 'numeric',
+          }),
+          authenticationFacilityName:
+            facilities[0]?.facilityInfo?.name || 'Unknown Facility',
+          authenticationFacilityID:
+            facilities[0]?.facilityInfo?.stationNumber || 'Unknown ID',
+        }
+      : {};
+
+  return {
+    authenticationSummary: authenticationInfo,
+    vaTreatmentFacilities: mappedFacilities,
+  };
+};
+
+export const blueButtonReducer = (state = initialState, action) => {
+  if (action.type === Actions.BlueButtonReport.GET) {
```

Comment:
For consistency, use the `switch` statement and just disable the linting for that line.

Reply:
No problem